### PR TITLE
Test firmware

### PR DESCRIPTION
**Pull request by InBrewJ**
Still need to:

- modularise test_discover within test_aux.sh
- investigate BAC0

(I made the PR to test Travis given the connectivity issues we're seeing with the cloud tests)

### CODE REVIEW

File: misc/test_site/mac_addrs/3c5ab41e8f0a/module_config.json [hunk 1]
```diff
@@ -10,7 +10,7 @@
     "name": "*** Name ***",
     "serial": "*** Serial ***",
     "type": "*** Type ***",
-    "firmware_version": "*** Firmware Version ***"
+    "firmware_version": "?\\xFF\\xFF\\x19,>u\\x08\\x00no"
```
Comment (ewRedstone) [inline misc/test_site/mac_addrs/3c5ab41e8f0a/module_config.json:13]: This doesnt look correct, looks as though some bytes are being inverted or not represented correctly as ascii

File: subset/pentests/test_discover [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+else
+    echo No module config $CONFIG
+fi
+
+# Sanity
+
+ping -n -c 3 $TARGET_IP
+
+echo Testing target $TARGET_IP port 23
+nc -nzv $TARGET_IP -w 5 23 || true
```
Comment (ewRedstone) [inline subset/pentests/test_discover]: same command duplicated three times

File: testing/test_aux.out [hunk 3]
```diff
@@ -77,6 +83,9 @@ port-02 module_config modules
   "brute": {
     "enabled": true
   },
+  "discover": {
```
Comment (ewRedstone) [inline testing/test_aux.out:86]: bizareley appearing twice

Comment (InBrewJ) [reply testing/test_aux.out:86]: https://github.com/arupiot/daq/blob/124084d4fdfb2773e8d6600f8f7b9d24d62319ec/testing/test_aux.out

It appears twice like all the other modules appear twice - those are the configs for 2 ports

File: misc/test_site/mac_addrs/3c5ab41e8f0a/module_config.json (hunk 1 above)
Comment (InBrewJ) [reply misc/test_site/mac_addrs/3c5ab41e8f0a/module_config.json:13]: That's the output I get from bacnet4j when querying it with nmap - I believe it's meant to be 1.0.1 - still trying to figure out what encoding it's in!

File: subset/pentests/test_discover (hunk 2 above)
Comment (InBrewJ) [reply subset/pentests/test_discover]: Fair!

**Pull request merged**